requests/adapters.py

```diff
@@ -392,7 +392,16 @@ def send(self, request, stream=False, timeout=None, verify=True, cert=None, prox
                         low_conn.send(b'\r\n')
                     low_conn.send(b'0\r\n\r\n')
 
-                    r = low_conn.getresponse()
+                    # Control timeout manually because we do not use
+                    # 'urlopen' but internal connection object.
+                    if hasattr(low_conn, 'sock'):
+                        low_conn.sock.settimeout(timeout.read_timeout)
+
+                    try:
+                        r = low_conn.getresponse()
+                    except (socket.error, socket.timeout) as err:
+                        conn._raise_timeout(err, url, timeout.read_timeout)
```

Comment:
Yeah me too. I have considered few options and I think this is the lesser evil. Only other option is to duplicate behaviour of `_raise_timeout` i.e. handling different compatibility quirks (like SSL timeouts, EAGAIN in py2, see: [code](https://github.com/kennethreitz/requests/blob/master/requests/packages/urllib3/connectionpool.py#L300-L315)). I think using this particular "private" method has a lesser introduces less of technology dept.

To be honest I think the only true and reasonable solution would be to fix the obvious deficiency of `urllib3` package, because once we did:
```
low_conn = conn._get_conn(timeout=DEFAULT_POOL_TIMEOUT)
```
there is no other way to do our work without calling `HTTPConnectionPool` internals. In fact the code section that starts from line 372 is like a big inlined excerpt from some non existing  `HTTPConnectionPool` method that should be some there.

I think this should be eventually handled in `urllib3` instead of here, but with existing dependencies policy in **requests** it is impossible to achieve that in single PR/patch.